File: app/src/main/java/com/wa2c/android/medoly/plugin/action/tweet/dialog/ConfirmDialogFragment.java
package com.wa2c.android.medoly.plugin.action.tweet.dialog;

import android.app.AlertDialog;
import android.app.Dialog;
import android.os.Bundle;
import android.text.TextUtils;


/**
 * 確認ダイアログを表示する。
 */
public class ConfirmDialogFragment extends AbstractDialogFragment {

    /** タイトルキー。 */
    private final static String ARG_TITLE = "TITLE";
    /** メッセージキー。 */
    private final static String ARG_MESSAGE = "MESSAGE";

    /** 標準ボタン判定。 */
    private final static String ARG_IS_BUTTON_DEFAULT = "IS_BUTTON_DEFAULT";
    /** Positiveボタンキー。 */
    private final static String ARG_POSITIVE_BUTTON = "POISITIVE_BUTTON";
    /** Neutralボタンキー。 */
    private final static String ARG_NEUTRAL_BUTTON = "NEUTRAL_BUTTON";
    /** Negativeボタンキー。 */
    private final static String ARG_NEGATIVE_BUTTON = "NEGATIVE_BUTTON";


    /**
     * ダイアログのインスタンスを作成する。(OK/Cancelボタン。)
     * @param message メッセージ。
     * @param title タイトル。
     * @return ダイアログのインスタンス。
     */
    static public ConfirmDialogFragment newInstance(CharSequence message, CharSequence title) {
        ConfirmDialogFragment fragment = new ConfirmDialogFragment();
        Bundle args = new Bundle();
        args.putCharSequence(ARG_MESSAGE, message);
        args.putCharSequence(ARG_TITLE, title);
        args.putBoolean(ARG_IS_BUTTON_DEFAULT, true);
        fragment.setArguments(args);

        return fragment;
    }

    /**
     * ダイアログのインスタンスを作成する。(ボタン文字指定。null指定の場合はボタン無し。)
     * @param message メッセージ。
     * @param title タイトル。
     * @param positiveButton Positiveボタン。
     * @param neutralButton  Neutralボタン。
     * @param negativeButton Negativeボタン。
     * @return ダイアログのインスタンス。
     */
    static public ConfirmDialogFragment newInstance(String message, String title, String positiveButton, String neutralButton, String negativeButton) {
        ConfirmDialogFragment fragment = new ConfirmDialogFragment();
        Bundle args = new Bundle();
        args.putCharSequence(ARG_MESSAGE, message);
        args.putCharSequence(ARG_TITLE, title);
        args.putString(ARG_POSITIVE_BUTTON, positiveButton);
        args.putString(ARG_NEUTRAL_BUTTON,  neutralButton);
        args.putString(ARG_NEGATIVE_BUTTON, negativeButton);
        args.putBoolean(ARG_IS_BUTTON_DEFAULT, false);
        fragment.setArguments(args);

        return fragment;
    }



    /**
     * onCreateDialogイベント処理。
     */
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        Bundle args = getArguments();

        // ダイアログビルダ
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
        builder.setTitle(args.getCharSequence(ARG_TITLE));
        builder.setMessage(args.getCharSequence(ARG_MESSAGE));

        // ボタン
        if (args.getBoolean(ARG_IS_BUTTON_DEFAULT)) {
            builder.setPositiveButton(android.R.string.ok, clickListener);
            builder.setNeutralButton(android.R.string.cancel, clickListener);
        } else {
            // Positiveボタン
            String positive = args.getString(ARG_POSITIVE_BUTTON);
            if (!TextUtils.isEmpty(positive)) {
                builder.setPositiveButton(positive, clickListener);
            }
            // Neutralボタン
            String neutral = args.getString(ARG_NEUTRAL_BUTTON);
            if (!TextUtils.isEmpty(neutral)) {
                builder.setNeutralButton(neutral, clickListener);

            }
            // Negativeボタン
            String negative = args.getString(ARG_NEGATIVE_BUTTON);
            if (!TextUtils.isEmpty(negative)) {
                builder.setNegativeButton(negative, clickListener);
            }
        }

        return  builder.create();
    }

}

File: app/src/main/java/com/wa2c/android/medoly/plugin/action/tweet/dialog/ConfirmDialogFragment.kt
package com.wa2c.android.medoly.plugin.action.tweet.dialog

import android.app.Dialog
import android.os.Bundle
import androidx.appcompat.app.AlertDialog

/**
 * Confirmation dialog
 */
class ConfirmDialogFragment : AbstractDialogFragment() {

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        super.onCreateDialog(savedInstanceState)
        val args = arguments!!

        // dialog build
        return AlertDialog.Builder(context).apply {
            setTitle(args.getCharSequence(ARG_TITLE))
            setMessage(args.getCharSequence(ARG_MESSAGE))

            // button
            if (args.getBoolean(ARG_IS_BUTTON_DEFAULT)) {
                setPositiveButton(android.R.string.ok, null)
                setNeutralButton(android.R.string.cancel, null)
            } else {
                // Positive button
                args.getString(ARG_POSITIVE_BUTTON)?.let {
                    if (it.isNotEmpty()) setPositiveButton(it, null)
                }
                // Neutral button
                args.getString(ARG_NEUTRAL_BUTTON)?.let {
                    if (it.isNotEmpty()) setNeutralButton(it, null)
                }
                // Negative button
                args.getString(ARG_NEGATIVE_BUTTON)?.let {
                    if (it.isNotEmpty()) setNegativeButton(it, null)
                }
            }
        }.create()
    }

    companion object {

        /** Title key.  */
        private const val ARG_TITLE = "TITLE"
        /** Message key. */
        private const val ARG_MESSAGE = "MESSAGE"


        private const val ARG_IS_BUTTON_DEFAULT = "IS_BUTTON_DEFAULT"
        /** Positive button key.  */
        private const val ARG_POSITIVE_BUTTON = "POSITIVE_BUTTON"
        /** Neutral button key.  */
        private const val ARG_NEUTRAL_BUTTON = "NEUTRAL_BUTTON"
        /** Negative button key.  */
        private const val ARG_NEGATIVE_BUTTON = "NEGATIVE_BUTTON"

        /**
         * Create dialog instance. (OK/Cancel)
         * @param message A message.
         * @param title A title.
         * @return New dialog instance.
         */
        fun newInstance(message: CharSequence, title: CharSequence?): ConfirmDialogFragment {
            return ConfirmDialogFragment().apply {
                arguments = Bundle().apply {
                    putCharSequence(ARG_MESSAGE, message)
                    putCharSequence(ARG_TITLE, title)
                    putBoolean(ARG_IS_BUTTON_DEFAULT, true)
                }
            }
        }

    }

}
